llvm/lib/Target/SPIRV/SPIRVEmitIntrinsics.cpp

```diff
@@ -265,6 +281,30 @@ Instruction *SPIRVEmitIntrinsics::visitBitCastInst(BitCastInst &I) {
   return NewI;
 }
 
+void SPIRVEmitIntrinsics::insertPtrCastInstr(Instruction *I) {
+  StoreInst *SI = dyn_cast<StoreInst>(I);
```

Comment:
As for now, I was able to only find cases where such casting is needed before stores. However, I might be missing some cases. Let me know if anyone has any other ideas.